demos/python_demos/action_recognition/action_recognition_demo/steps.py

```diff
@@ -41,25 +41,37 @@ def run_pipeline(video, encoder, decoder, render_fn, decoder_seq_size=16, fps=30
 
 class DataStep(PipelineStep):
 
-    def __init__(self, video_list, loop=True):
+    def __init__(self, video_filepath, loop):
         super().__init__()
-        self.video_list = video_list
+        self.video = video_filepath
         self.cap = None
+        self.loop = loop
 
-        if loop:
-            self._video_cycle = cycle(self.video_list)
+        if self.loop:
+            self._video_cycle = cycle(self.video)
         else:
-            self._video_cycle = iter(self.video_list)
+            self._video_cycle = iter(self.video)
+
+        try:
+            self._open_video()
+            if not self.loop:
+                self._video_cycle = iter(self.video)
```

Comment:
You should open the video only once. You won't need to iterate over `self.video`.